libshaderc/include/shaderc.h

```diff
@@ -105,6 +111,11 @@ void shaderc_compile_options_release(shaderc_compile_options_t options);
 void shaderc_compile_options_add_macro_definition(
     shaderc_compile_options_t options, const char* name, const char* value);
 
+// Set flags for the rules on which warnings or errors will be issued. This can
+// either be an individual shaderc_msg_rules flag, or a set of flags or'd together
+void shaderc_compile_options_set_message_rules(
```

Comment:
Correspondingly, should be 
```
// Sets the target environment and version, overriding the default or any previous call to this
// method. Default is shaderc_target_env_vulkan.
void shader_compile_options_set_target_env(..., shaderc_target_env);
```

The target can be an enum, since they are exclusive choices.
(If it were a bit mask, then we'd want an unsigned int to convey it.)

I'm waffling over whether we should also have a version parameter to state which version of the target spec sets the rules.  So far no OpenGL spec sets rules for SPIR-V, but has an established numbering convention such as 450 for OpenGL 4.5.   Vulkan is still under development, but later versions might have different rules than what goes into Vulkan 1.0.   For now I think we can address any future movement in the rules with a separate shader_compile_options_set_target_env_version(..., int version);  if we need it.